airflow/assets/__init__.py

```diff
@@ -372,6 +373,24 @@ def iter_dag_dependencies(self, *, source: str, target: str) -> Iterator[DagDepe
         )
 
 
+class Dataset(Asset):
+    """A representation of dataset dependencies between workflows."""
+
+    asset_type: ClassVar[str] = "dataset"
+
+    def __init__(self, *args, group="dataset", **kwargs) -> None:
+        super().__init__(*args, group=group, **kwargs)
```

Comment:
Maybe an easier way to do this would be to add `asset_type = ""` on `Asset`, and have `group` default to `asset_type` in Asset’s `__init__`? That would automatically make user-defined subclasses work better.

Reply:
Updated